### PR TITLE
grafana-cli: Fix file path processing, returning of errors

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Fix grafana-cli so it uses `filepath.Join` for filesystem path joining, so it works on Windows. Also fix some returning of errors.

### CODE REVIEW

**Review by papagian — APPROVED** [review]
LGTM :heart:

I wonder how we haven't encountered any issues on Windows earlier. I guess not a lot of users use customised plugins in Windows?

File: pkg/cmd/grafana-cli/commands/install_command.go [hunk 1]
```diff
@@ -225,27 +224,26 @@ func extractFiles(archiveFile string, pluginName string, filePath string, allowS
...
-				err = extractSymlink(zf, newFile)
-				if err != nil {
+				if err := extractSymlink(zf, newFile); err != nil {
 					logger.Errorf("Failed to extract symlink: %v \n", err)
 					continue
 				}
 			} else {
-				err = extractFile(zf, newFile)
-				if err != nil {
+				if err := extractFile(zf, newFile); err != nil {
```
Comment (papagian) [inline pkg/cmd/grafana-cli/commands/install_command.go:246]: lint prefers `else if` here

Comment (aknuds1) [reply pkg/cmd/grafana-cli/commands/install_command.go:246]: @papagian Ah argh.

**Comment by aknuds1** [comment]
> I wonder how we haven't encountered any issues on Windows earlier. I guess not a lot of users use customised plugins in Windows?

@papagian Wondering the same...

File: pkg/cmd/grafana-cli/commands/install_command.go (hunk 1 above)
Comment (aknuds1) [reply pkg/cmd/grafana-cli/commands/install_command.go:246]: Will fix.

**Comment by hugohaggmark** [comment]
> > I wonder how we haven't encountered any issues on Windows earlier. I guess not a lot of users use customised plugins in Windows?
> 
> @papagian Wondering the same...

What was the issue on Windows?

**Comment by papagian** [comment]
> What was the issue on Windows?

Assuming `filePath` is `C:\grafana\data\plugins\plugin` and `newFile` is `something` then `path.Join(filePath, newFileName)` becomes `C:\grafana\data\plugins\plugin/something` which is not valid path for Windows. Check [here](https://stackoverflow.com/questions/39181790/whats-the-difference-between-path-path-filepath-packages-in-go#answer-39182128) for more details.

**Comment by aknuds1** [comment]
@hugohaggmark @papagian is correct. We can't use `path.Join` for filesystem paths, since it assumes the path separator is '/'. The `path` package is meant for URI paths, not filesystem ones.

**Comment by hugohaggmark** [comment]
Thank you @papagian @aknuds1 🙏

**Pull request merged**